src/coreclr/jit/compiler.h

```diff
@@ -1099,14 +1099,14 @@ class LclVarDsc
     {
         return varTypeIsSmall(TypeGet()) &&
                // lvIsStructField is treated the same as the aliased local, see fgDoNormalizeOnStore.
-               (lvIsParam || m_addrExposed || lvIsStructField);
+               (lvIsParam || m_addrExposed || lvIsStructField || lvIsOSRLocal);
     }
 
     bool lvNormalizeOnStore() const
     {
         return varTypeIsSmall(TypeGet()) &&
                // lvIsStructField is treated the same as the aliased local, see fgDoNormalizeOnStore.
-               !(lvIsParam || m_addrExposed || lvIsStructField);
+               !(lvIsParam || m_addrExposed || lvIsStructField || lvIsOSRLocal);
```

Comment:
It took me a while to guess that the reason is that the local could have been marked normalize-on-load in tier-0 due to the more conservative address exposure there. Maybe add a comment to that effect?

Also, I guess if we wanted to we could scope this down to just OSR locals that were address exposed in tier 0, since we have that information available.

Reply:
I thought about scoping it down, so let me do that (will need a new bit on lclvar dsc).